src/funjson.c

```diff
@@ -160,20 +191,17 @@ FUNCTION(fun_json_query)
     }
     break;
   case JSON_QUERY_UNESCAPE: {
-    char *latin1, *c;
+    char *latin1;
     int len;
     if (!cJSON_IsString(json)) {
       safe_str("#-1", buff, bp);
       break;
     }
-    latin1 =
-      utf8_to_latin1(cJSON_GetStringValue(json), -1, &len, 0, "json.string");
-    for (c = latin1; *c; c += 1) {
-      if (!isprint(*c) && !isspace(*c)) {
-        *c = '?';
-      }
-    }
-    safe_strl(latin1, len, buff, bp);
+
+    latin1 = utf8_to_latin1(cJSON_GetStringValue(json), -1, &len, 0, "json.string");
+    char * partialbuf = json_unescape_latin1string(latin1,len);
```

Comment:
Variable should be defined at the start of the block with the others for wider compatability

Reply:
Doing this -- but for my knowledgebase, what does this do to compatibility?